--- printer_ui-orig/printerui/app_de.ts
+++ printer_ui/printerui/app_de.ts
@@ -3435,4 +3435,229 @@
         <translation>Sind Sie sich sicher, dass sie diesen Druck abbrechen wollen?</translation>
     </message>
 </context>
+<context>
+  <name>BootOptionsPage</name>
+  <message>
+    <source>Cancel</source>
+    <translation>Abbrechen</translation>
+  </message>
+  <message>
+    <source>Resetting X1Plus settings can help to recover from modified X1Plus installations.  </source>
+    <translation>Das Zurücksetzen der X1Plus-Einstellungen kann helfen, sich von fehlerhaften X1Plus-Modifikationen zu erholen.  </translation>
+  </message>
+  <message>
+    <source>It will also reset the lock screen passcode, root password, and ssh keys.  (This process takes a few minutes.)</source>
+    <translation>Dies wird auch das Sperrbildschirm-Passwort, root-Passwort und die SSH-Schlüssel zurücksetzen. (Dieser Vorgang kann einige Minuten dauern)</translation>
+  </message>
+  <message>
+    <source>The emergency recovery console will connect the printer to its default WiFi network and start a temporary SSH server as root. It will also spawn a root console on the AP board's UART port and USB port.</source>
+    <translation>Die Notfallwiederherstelungskonsole wird den Drucker mit dem Standard WLAN Netzwerk verbinden und einen temporären SSH Server starten. Es startet auch eine root Konsole auf dem UART- und USB Port der AP Platine.</translation>
+  </message>
+  <message>
+    <source>It is possible to unrecoverably and permanently damage your printer with a root console. Never type something into a root console unless you understand what you're typing.</source>
+    <translation>Es ist möglich, Ihren Drucker unwiederbringlich und dauerhaft mit einer Root-Konsole zu beschädigen. Geben sie niemals etwas in eine Root-Konsole ein, wenn sie nicht genau wissen was sie machen. Ich meine wirklich genau wissen was sie machen.</translation>
+  </message>
+  <message>
+    <source>I'll be careful, I promise</source>
+    <translation>Ich verspreche ich werde vorsichtig sein</translation>
+  </message>
+  <message>
+    <source>Starting from the built-in firmware can be useful to diagnose issues with X1Plus or with your printer. This option attempts to enter LAN mode before starting, and disables the printer's internal upgrade service; you can use this option to help avoid inadvertent erasure of your X1Plus installation.</source>
+    <translation>Von der original Firmware zu starten kann sinnvoll sein, um Probleme mit X1Plus oder mit dem Drucker zu identifizieren. Diese Option probiert vor dem Start den "nur LAN"-Modus zu aktivieren. Außerdem wird der interne Updateservice deaktiviert. Sie können diese Option benutzen um zu verhindern, dass sie versehentlich ihre X1Plus Installation löschen.</translation>
+  </message>
+  <message>
+    <source>Fingers are crossed</source>
+    <translation>Die Daumen sind gedrückt</translation>
+  </message>
+  <message>
+    <source>&lt;b&gt;Start X1Plus installer&lt;/b&gt;</source>
+    <translation>&lt;b&gt;X1Plus installieren&lt;/b&gt;</translation>
+  </message>
+  <message>
+    <source>Reset X1Plus settings</source>
+    <translation>X1Plus-Einstellungen zurücksetzen</translation>
+  </message>
+  <message>
+    <source>Emergency recovery console</source>
+    <translation>Notfallwiederherstellungskonsole</translation>
+  </message>
+  <message>
+    <source>X1Plus startup options</source>
+    <translation>X1Plus Startoptionen</translation>
+  </message>
+  <message>
+    <source>Use these tools to upgrade or repair the X1Plus firmware on your printer.&lt;br&gt;&lt;br&gt;&lt;font color="#EEAAAA"&gt;&lt;b&gt;Starting up using on-device firmware can result in your printer losing access to custom firmware.  Do this only as a last resort.&lt;/b&gt;&lt;/font&gt;</source>
+    <translation>Verwenden Sie diese Werkzeuge, um die X1Plus-Firmware auf Ihrem Drucker zu aktualisieren oder zu reparieren.&lt;br&gt;&lt;br&gt;&lt;font color="#EEAAAA"&gt;&lt;b&gt;Starten der On-Device-Firmware kann dazu führen, dass Ihr Drucker den Zugriff auf die X1Plus Firmware verliert. Dies nur als letzten Ausweg tun.&lt;/b&gt;&lt;/font&gt;</translation>
+  </message>
+</context>
+<context>
+  <name>HardwarePage</name>
+  <message>
+    <source>Not installed</source>
+    <translation>Nicht installiert</translation>
+  </message>
+  <message>
+    <source>Installed</source>
+    <translation>Installiert</translation>
+  </message>
+  <message>
+    <source>Lights</source>
+    <translation>Lampen</translation>
+  </message>
+</context>
+<context>
+  <name>BedTramming</name>
+  <message>
+    <source>Step 1:</source>
+    <translation>Schritt 1:</translation>
+  </message>
+  <message>
+    <source>Step 2:</source>
+    <translation>Schritt 2:</translation>
+  </message>
+  <message>
+    <source>Step 3:</source>
+    <translation>Schritt 3:</translation>
+  </message>
+  <message>
+    <source>Step 4:</source>
+    <translation>Schritt 4:</translation>
+  </message>
+  <message>
+    <source>Step 5:</source>
+    <translation>Schritt 5:</translation>
+  </message>
+  <message>
+    <source>Prepare</source>
+    <translation>Vorbereiten</translation>
+  </message>
+</context>
+<context>
+  <name>VibrationCompPlotter</name>
+  <message>
+    <source>X axis</source>
+    <translation>X-Achse</translation>
+  </message>
+  <message>
+    <source>Y axis</source>
+    <translation>Y-Achse</translation>
+  </message>
+</context>
+<context>
+  <name>InstallingPage</name>
+  <message>
+    <source>Yes</source>
+    <translation>Ja</translation>
+  </message>
+  <message>
+    <source>No</source>
+    <translation>Nein</translation>
+  </message>
+</context>
+<context>
+  <name>BedMesh</name>
+  <message>
+    <source>Bed mesh calibration: Toolhead is homing</source>
+    <translation>Höhenprofil Kalibrierung: Werkzeugkopf wird referenziert</translation>
+  </message>
+</context>
+<context>
+  <name>VibrationComp</name>
+  <message>
+    <source>X axis</source>
+    <translation>X-Achse</translation>
+  </message>
+  <message>
+    <source>Y axis</source>
+    <translation>Y-Achse</translation>
+  </message>
+  <message>
+    <source>Range:</source>
+    <translation>Bereich:</translation>
+  </message>
+  <message>
+    <source>to</source>
+    <translation>bis</translation>
+  </message>
+  <message>
+    <source>Hz</source>
+    <translation>Hz</translation>
+  </message>
+</context>
+<context>
+  <name>EmergencyConsole</name>
+  <message>
+    <source>It is possible to do permanent, irreversible damage to your printer from a root console. Do not enter commands unless you understand what you are typing.</source>
+    <translation>Es ist möglich, Ihren Drucker unwiederbringlich und dauerhaft mit einer Root-Konsole zu beschädigen. Geben sie niemals etwas in eine Root-Konsole ein, wenn sie nicht genau wissen was sie machen.</translation>
+  </message>
+</context>
+<context>
+  <name>ConsolePage</name>
+  <message>
+    <source>This interface allows you to send G-code commands to the printer. You can enter commands with the virtual keyboard, or put together commands from the shortcut bar at the bottom of the screen. The printer's G-code parser is somewhat picky; here are some tips for how to placate it: </source>
+    <translation>Diese Schnittstelle elaubt es dir G-Code-Befehle an den Drucker zu senden. Sie können Befehle mit der virtuellen Tastatur eingeben oder Befehle aus den Vorgaben am unteren Rand des Bildschirms zusammenstellen. Der G-Code Parser des Druckers ist etwas pingelig; hier sind einige Tipps, wie man es formuliert: </translation>
+  </message>
+  <message>
+    <source>M106: G-code command for fan control</source>
+    <translation>M106: G-Code Befehl zur Lüftersteuerung</translation>
+  </message>
+  <message>
+    <source>S255: parameter to set fan speed (0 to 255)</source>
+    <translation>S255: Parameter zum Setzen der Lüftergeschwindigkeit (0 bis 255)</translation>
+  </message>
+  <message>
+    <source>M106 P2 S255
+G4 S5
+M106 P2 S0</source>
+    <translation>M106 P2 S255
+G4 S5
+M106 P2 S0</translation>
+  </message>
+  <message>
+    <source>This interface allows you to run commands on your printer as root. You can enter commands with the virtual keyboard, or put together commands from the shortcut bar at the bottom of the screen. Commands are executed synchronously, so long-running commands or commands that require user input may hang the UI; use caution! This is intended as a quick diagnostic tool, but for more intensive tasks, consider SSHing to the printer instead.</source>
+    <translation>Mit dieser Schnittstelle können Sie Befehle auf Ihrem Drucker als root ausführen. Sie können Befehle mit der virtuellen Tastatur eingeben oder Befehle aus den Vorgaben am unteren Rand des Bildschirms zusammenstellen. Befehle werden synchron ausgeführt, so dass lange laufende Befehle oder Befehle, die Benutzereingaben erfordern, die UI aufhängen; Vorsicht! Dies ist als Schnelldiagnosewerkzeug gedacht, aber für intensivere Aufgaben sollten Sie stattdessen per SSH anmelden.</translation>
+  </message>
+  <message>
+    <source>WARNING: It is possible to do permanent, irreversible damage to your printer from a root console. Do not enter commands unless you understand what you are typing.</source>
+    <translation>Warnung: Es ist möglich, Ihren Drucker unwiederbringlich und dauerhaft mit einer Root-Konsole zu beschädigen. Geben sie keine Befehle ein, wenn sie nicht genau wissen was sie machen.</translation>
+  </message>
+  <message>
+    <source>enter a G-code command</source>
+    <translation>gebe einen G-Code Befehl ein</translation>
+  </message>
+  <message>
+    <source>enter a shell command to run as root</source>
+    <translation>gib ein Kommmando ein, welches als root ausgeführt wird</translation>
+  </message>
+  <message>
+    <source>Printer is running! Cannot execute gcode now</source>
+    <translation>Der Drucker läuft gerade! G-Code kann jetzt nicht ausgeführt werden</translation>
+  </message>
+</context>
+<context>
+  <name>DeviceInfo</name>
+  <message>
+    <source>Return</source>
+    <translation>Zurück</translation>
+  </message>
+  <message>
+    <source>Total print time: </source>
+    <translation>Druckzeit insgesamt: </translation>
+  </message>
+  <message>
+    <source>AMS total filament switches: </source>
+    <translation>AMS Anzahl der Filamentwechsel: </translation>
+  </message>
+  <message>
+    <source>AMS total failed filament switches: </source>
+    <translation>AMS Anzahl der fehlerhaften Filamentwechsel: </translation>
+  </message>
+</context>
+<context>
+  <name>ScreenLockPage</name>
+  <message>
+    <source>Return</source>
+    <translation>Zurück</translation>
+  </message>
+</context>
 </TS>
\ No newline at end of file
