--- printer_ui-orig/printerui/app_sv.ts
+++ printer_ui/printerui/app_sv.ts
@@ -3436,4 +3436,635 @@
         <translation>Är du säker att du vill avbryta utskriften?</translation>
     </message>
 </context>
+<context>
+  <name>EmergencyConsole</name>
+  <message>
+    <source>SSH to %1.</source>
+    <translation>SSH till %1.</translation>
+  </message>
+  <message>
+    <source>&lt;b&gt;Connect a micro-USB cable to the AP board&lt;/b&gt; and use ADB to connect.</source>
+    <translation>&lt;b&gt;Anslut en mikro-USB-kabel till AP-kortet&lt;/b&gt; och använd ADB för att ansluta.</translation>
+  </message>
+  <message>
+    <source>Connect a UART to the AP board.</source>
+    <translation>Anslut en UART till AP-kortet.</translation>
+  </message>
+  <message>
+    <source>It is possible to do permanent, irreversible damage to your printer from a root console. Do not enter commands unless you understand what you are typing.</source>
+    <translation>Det är möjligt att göra permanenta, oåterkalleliga skador på din skrivare från en rotkonsol. Ange inte kommandon om du inte förstår vad du skriver.</translation>
+  </message>
+</context>
+<context>
+  <name>SelectX1pPage</name>
+  <message>
+    <source>Select a firmware bundle to install:</source>
+    <translation>Välj en firmware-bunt att installera:</translation>
+  </message>
+  <message>
+    <source>Are you sure you're ready to install the X1Plus custom firmware?&lt;br&gt;&lt;br&gt;While we've done our best to make X1Plus as safe as possible, there are always risks associated with modifying your printer. X1Plus comes with NO WARRANTY, EXPRESS OR IMPLIED, and by installing it, you accept that you and you alone are responsible for the risk of damage to your printer, or its surroundings.&lt;br&gt;&lt;br&gt;Once the installation process begins, don't touch your printer until it completes. It could take up to 10 minutes (or longer, depending on Internet connection speed) to install.</source>
+    <translation>Är du säker på att du'är redo att installera X1Plus anpassad firmware?&lt;br&gt;&lt;br&gt;Medan vi'har gjort vårt bästa för att göra X1Plus så säkert som möjligt, det finns alltid risker i samband med att ändra din skrivare. X1Plus levereras med INGEN GARANTI, EXPRESS ELLER IMPLIED, och genom att installera det, du accepterar att du och du ensam är ansvariga för risken för skador på din skrivare eller dess omgivning.&lt;br&gt;&lt;br&gt;När installationen påbörjas, rör don't din skrivare tills den är klar. Det kan ta upp till 10 minuter (eller längre, beroende på Internet-anslutning hastighet) att installera.</translation>
+  </message>
+  <message>
+    <source>Er, never mind</source>
+    <translation>Er, aldrig emot</translation>
+  </message>
+</context>
+<context>
+  <name>X1PlusOTADialog</name>
+  <message>
+    <source>Return</source>
+    <translation>Tillbaka</translation>
+  </message>
+</context>
+<context>
+  <name>UpgradeDialog</name>
+  <message>
+    <source>Download %1</source>
+    <translation>Ladda ner %1</translation>
+  </message>
+  <message>
+    <source>Install %1</source>
+    <translation>Installera %1</translation>
+  </message>
+  <message>
+    <source>Cancel</source>
+    <translation>Avbryt</translation>
+  </message>
+  <message>
+    <source>Return</source>
+    <translation>Retur</translation>
+  </message>
+  <message>
+    <source>%1 update %2</source>
+    <translation>%1 uppdatering %2</translation>
+  </message>
+  <message>
+    <source>Version %1 is available on the SD card and ready to install. Install it now?</source>
+    <translation>Version %1 finns på SD-kortet och är klar att installera. Installera det nu?</translation>
+  </message>
+  <message>
+    <source>Installing version %1.  Do not power off your printer.</source>
+    <translation>Installera version %1. Stäng inte av skrivaren.</translation>
+  </message>
+  <message>
+    <source>Successfully installed version %1.</source>
+    <translation>Installationen av version %1 lyckades.</translation>
+  </message>
+  <message>
+    <source> (Updater message "%1".)</source>
+    <translation> (Uppdateringsmeddelande "%1".)</translation>
+  </message>
+</context>
+<context>
+  <name>MeshCalcs</name>
+  <message>
+    <source>Your hotbed appears to be flat and level.</source>
+    <translation>Din bädd verkar vara platt och nivå.</translation>
+  </message>
+</context>
+<context>
+  <name>ScreenLockPage</name>
+  <message>
+    <source>Return</source>
+    <translation>Retur</translation>
+  </message>
+  <message>
+    <source>2 minutes</source>
+    <translation>2 minuter</translation>
+  </message>
+  <message>
+    <source>5 minutes</source>
+    <translation>5 minuter</translation>
+  </message>
+  <message>
+    <source>10 minutes</source>
+    <translation>10 minuter</translation>
+  </message>
+  <message>
+    <source>15 minutes</source>
+    <translation>15 minuter</translation>
+  </message>
+  <message>
+    <source>Time before display sleeps</source>
+    <translation>Tid innan skärmen sover</translation>
+  </message>
+  <message>
+    <source>Lock screen mode</source>
+    <translation>Lås skärmläge</translation>
+  </message>
+  <message>
+    <source>Screen saver only</source>
+    <translation>Endast skärmsläckare</translation>
+  </message>
+  <message>
+    <source>Swipe to unlock</source>
+    <translation>Svep för att låsa upp</translation>
+  </message>
+  <message>
+    <source>Passcode: "%1"</source>
+    <translation>Lösenord: "%1"</translation>
+  </message>
+  <message>
+    <source>Passcode is unset</source>
+    <translation>Lösenordet är inaktiverat</translation>
+  </message>
+  <message>
+    <source>Change passcode</source>
+    <translation>Ändra lösenord</translation>
+  </message>
+  <message>
+    <source>You can add custom text or a background image to the lock screen by creating one or both of the following files on the SD card:&lt;br&gt;&lt;br&gt;&lt;b&gt;Text&lt;/b&gt;: /x1plus/lockscreen.txt&lt;br&gt;&lt;b&gt;Image&lt;/b&gt;: /x1plus/lockscreen.png</source>
+    <translation>Du kan lägga till anpassad text eller bakgrundsbild till låsskärmen genom att skapa en eller båda av följande filer på SD-kortet:&lt;br&gt;&lt;br&gt;&lt;b&gt;Text&lt;/b&gt;: /x1plus/lockscreen xt&lt;br&gt;&lt;b&gt;bild&lt;/b&gt;: /x1plus/lockscreen.png</translation>
+  </message>
+</context>
+<context>
+  <name>VibrationComp</name>
+  <message>
+    <source>Vibration Compensation</source>
+    <translation>Vibrationskompensering</translation>
+  </message>
+  <message>
+    <source>Beginning vibration compensation calibration.</source>
+    <translation>Börjar vibrationskompensation kalibrering.</translation>
+  </message>
+  <message>
+    <source>X axis</source>
+    <translation>X axel</translation>
+  </message>
+  <message>
+    <source>Y axis</source>
+    <translation>Y axel</translation>
+  </message>
+  <message>
+    <source>Sweeping %1
+%2 Hz: %3 dB
+Progress: %4%</source>
+    <translation>Svepande %1
+%2 Hz: %3 dB
+Förlopp: %4%</translation>
+  </message>
+  <message>
+    <source>An error has occurred and data collection timed out.</source>
+    <translation>Ett fel har inträffat och datainsamlingen har gått ut.</translation>
+  </message>
+  <message>
+    <source>Vibration compensation calibration: internal state error, status = %1</source>
+    <translation>Vibrationskompensation kalibrering: internt tillståndsfel, status = %1</translation>
+  </message>
+  <message>
+    <source>Vibration</source>
+    <translation>Vibration</translation>
+  </message>
+  <message>
+    <source>Run frequency sweep</source>
+    <translation>Kör frekvens svep</translation>
+  </message>
+  <message>
+    <source>to</source>
+    <translation>till</translation>
+  </message>
+  <message>
+    <source>Hz</source>
+    <translation>Hz</translation>
+  </message>
+  <message>
+    <source>Calibration History</source>
+    <translation>Kalibreringshistorik</translation>
+  </message>
+  <message>
+    <source>Clear calibration logs</source>
+    <translation>Rensa kalibreringsloggar</translation>
+  </message>
+  <message>
+    <source>This will remove all of your saved vibration compensation data.  Are you sure you want to proceed?</source>
+    <translation>Detta kommer att ta bort alla dina sparade vibrationsersättningsdata. Är du säker på att du vill fortsätta?</translation>
+  </message>
+</context>
+<context>
+  <name>GpioKeys</name>
+  <message>
+    <source>Sleep/wake</source>
+    <translation>Vila/vakna</translation>
+  </message>
+  <message>
+    <source>Ignore</source>
+    <translation>Ignorera</translation>
+  </message>
+</context>
+<context>
+  <name>BedMesh</name>
+  <message>
+    <source>&lt;font size="3" color="#AEAEAE"&gt;%1&lt;br&gt;&lt;br&gt;Calculated bed tilt (X-axis) is %2 mm across the X axis, and %3 mm across the Y axis; post-tramming peak-to-peak deviation is %4 mm.&lt;/font&gt;</source>
+    <translation>&lt;font size="3" color="#AEAEAE"&gt;%1&lt;br&gt;&lt;br&gt;Beräknad bäddlutning (X-axel) är %2 mm över X-axeln, och %3 mm över Y-axeln; Efter spårning topp-till-topp-avvikelse är %4 mm.&lt;/font&gt;</translation>
+  </message>
+  <message>
+    <source>Bed mesh calibration: Toolhead is homing</source>
+    <translation>Kalibrering av sängnät: Verktygshuvudet är hemmabruk</translation>
+  </message>
+  <message>
+    <source>Probed point %1&lt;br&gt;X=%2, Y=%3&amp;nbsp;&amp;nbsp;Z=%4 mm
+Progress: %5%</source>
+    <translation>Försedd punkt %1&lt;br&gt;X=%2, Y=%3&amp;nbsp;&amp;nbsp;Z=%4 mm
+Förlopp: %5%</translation>
+  </message>
+  <message>
+    <source>An error has occurred and data collection timed out.</source>
+    <translation>Ett fel har inträffat och datainsamlingen har gått ut.</translation>
+  </message>
+  <message>
+    <source>Bed mesh calibration: internal state error, status = %1</source>
+    <translation>Kalibrering av sängnät: internt tillståndsfel, status = %1</translation>
+  </message>
+  <message>
+    <source>Run diagnostics</source>
+    <translation>Kör diagnostik</translation>
+  </message>
+  <message>
+    <source>Tram bed</source>
+    <translation>Spårvagn säng</translation>
+  </message>
+  <message>
+    <source>Calibration History</source>
+    <translation>Kalibreringshistorik</translation>
+  </message>
+</context>
+<context>
+  <name>HardwarePage</name>
+  <message>
+    <source>Stainless steel</source>
+    <translation>Rostfritt stål</translation>
+  </message>
+  <message>
+    <source>Hardened steel</source>
+    <translation>Härdat stål</translation>
+  </message>
+  <message>
+    <source>Not installed</source>
+    <translation>Inte installerad</translation>
+  </message>
+  <message>
+    <source>Installed</source>
+    <translation>Installerad</translation>
+  </message>
+  <message>
+    <source>Accessories</source>
+    <translation>Tillbehör</translation>
+  </message>
+  <message>
+    <source>Auxiliary Part Cooling Fan</source>
+    <translation>Hjälpmedel del kylfläkt</translation>
+  </message>
+  <message>
+    <source>Interface</source>
+    <translation>Gränsgränssnitt</translation>
+  </message>
+  <message>
+    <source>Reset button mapping to default settings?</source>
+    <translation>Återställ knappmappning till standardinställningar?</translation>
+  </message>
+  <message>
+    <source>Buttons</source>
+    <translation>Knappar</translation>
+  </message>
+  <message>
+    <source>Short Press</source>
+    <translation>Kort tryck</translation>
+  </message>
+  <message>
+    <source>Long Press</source>
+    <translation>Långtryck</translation>
+  </message>
+  <message>
+    <source>Lights</source>
+    <translation>Lampor</translation>
+  </message>
+  <message>
+    <source>Backlight</source>
+    <translation>Bakgrundsbelysning</translation>
+  </message>
+  <message>
+    <source>Toolhead LED</source>
+    <translation>Verktygshuvud LED</translation>
+  </message>
+  <message>
+    <source>If you have replaced the accessories, please update the settings below to keep it consistent with the actual condition of the accessory to ensure the print quality</source>
+    <translation>Om du har ersatt tillbehören, uppdatera inställningarna nedan för att hålla den i överensstämmelse med det faktiska skicket på tillbehöret för att säkerställa utskriftskvaliteten</translation>
+  </message>
+</context>
+<context>
+  <name>DeviceInfo</name>
+  <message>
+    <source>%1: Device Info</source>
+    <translation>%1: Enhetsinformation</translation>
+  </message>
+  <message>
+    <source>Return</source>
+    <translation>Retur</translation>
+  </message>
+  <message>
+    <source>Printer serial number: </source>
+    <translation>Serienummer för skrivaren: </translation>
+  </message>
+  <message>
+    <source>Total print time: </source>
+    <translation>Total utskriftstid: </translation>
+  </message>
+  <message>
+    <source>%1 hours</source>
+    <translation>%1 timmar</translation>
+  </message>
+  <message>
+    <source>reset</source>
+    <translation>återställ</translation>
+  </message>
+  <message>
+    <source>Lead screws last lubricated: </source>
+    <translation>Blyskruvar senaste smörja: </translation>
+  </message>
+  <message>
+    <source>ASA carbon rod maintenance timer: </source>
+    <translation>ASA kolstång underhållstider: </translation>
+  </message>
+  <message>
+    <source>Bed mesh calibrated: </source>
+    <translation>Sängnät kalibrerat: </translation>
+  </message>
+  <message>
+    <source>Vibration compensation calibrated: </source>
+    <translation>Vibrationsersättning kalibrerad: </translation>
+  </message>
+  <message>
+    <source>AMS total length: </source>
+    <translation>AMS totala längd: </translation>
+  </message>
+  <message>
+    <source>%1 meters</source>
+    <translation>%1 meter</translation>
+  </message>
+  <message>
+    <source>AMS total print time: </source>
+    <translation>AMS totala utskriftstid: </translation>
+  </message>
+  <message>
+    <source>AMS total filament switches: </source>
+    <translation>AMS totala glödbrytare: </translation>
+  </message>
+  <message>
+    <source>Tray %1 print time: </source>
+    <translation>Utskriftstid för fack %1: </translation>
+  </message>
+  <message>
+    <source>Tray %1 print length: </source>
+    <translation>Bricka %1 utskriftslängd: </translation>
+  </message>
+  <message>
+    <source>%1 - print time: </source>
+    <translation>%1 - utskriftstid: </translation>
+  </message>
+  <message>
+    <source>%1 - print length: </source>
+    <translation>%1 - utskriftslängd: </translation>
+  </message>
+</context>
+<context>
+  <name>KexecDialog</name>
+  <message>
+    <source>Start from SD card</source>
+    <translation>Börja från SD-kort</translation>
+  </message>
+  <message>
+    <source>Your printer will automatically boot from the SD card in %1 seconds.</source>
+    <translation>Din skrivare startar automatiskt från SD-kortet om %1 sekunder.</translation>
+  </message>
+  <message>
+    <source>Your printer is rebooting into the OS on the inserted SD card.</source>
+    <translation>Din skrivare startar om till OS på det infogade SD-kortet.</translation>
+  </message>
+</context>
+<context>
+  <name>ScreenLock</name>
+  <message>
+    <source>This printer is locked.</source>
+    <translation>Denna skrivare är låst.</translation>
+  </message>
+</context>
+<context>
+  <name>BedTramming</name>
+  <message>
+    <source>Step 1:</source>
+    <translation>Steg 1:</translation>
+  </message>
+  <message>
+    <source>Step 2:</source>
+    <translation>Steg 2:</translation>
+  </message>
+  <message>
+    <source>Tap the "Prepare" button below to home the bed.</source>
+    <translation>Tryck på "Förbered" knappen nedan för att hem sängen.</translation>
+  </message>
+  <message>
+    <source>Step 3:</source>
+    <translation>Steg 3:</translation>
+  </message>
+  <message>
+    <source>Tap the "1" button to move the nozzle to the front left position.  Adjust the front left bed adjustment screw until the nozzle just barely touches the bed.</source>
+    <translation>Tryck på "1" knappen för att flytta munstycket till vänster fram. Justera den främre vänstra sängen justering skruv tills munstycket bara knappt vidrör sängen.</translation>
+  </message>
+  <message>
+    <source>Step 4:</source>
+    <translation>Steg 4:</translation>
+  </message>
+  <message>
+    <source>Adjust the front right and rear positions by tapping their corresponding button and adjusting their corresponding screws.</source>
+    <translation>Justera de främre högra och bakre positionerna genom att trycka på motsvarande knapp och justera motsvarande skruvar.</translation>
+  </message>
+  <message>
+    <source>Step 5:</source>
+    <translation>Steg 5:</translation>
+  </message>
+  <message>
+    <source>Check each position in order, repeating all three until no further adjustment is needed.</source>
+    <translation>Kontrollera varje position i ordning, upprepa alla tre tills ingen ytterligare justering behövs.</translation>
+  </message>
+  <message>
+    <source>Bed Tramming</source>
+    <translation>Spårning av sängkläder</translation>
+  </message>
+  <message>
+    <source>Prepare</source>
+    <translation>Förbered</translation>
+  </message>
+  <message>
+    <source>Rear
+   2</source>
+    <translation>Bakre
+   2</translation>
+  </message>
+  <message>
+    <source>Front L
+     1</source>
+    <translation>Framsida L
+     1</translation>
+  </message>
+  <message>
+    <source>Front R
+     3</source>
+    <translation>Främre R
+     3</translation>
+  </message>
+</context>
+<context>
+  <name>BedLevelDiag</name>
+  <message>
+    <source>Are you sure you want to remove the data entry from $1?</source>
+    <translation>Är du säker på att du vill ta bort datainmatningen från $1?</translation>
+  </message>
+</context>
+<context>
+  <name>VersionDialog</name>
+  <message>
+    <source>Install %1</source>
+    <translation>Installera %1</translation>
+  </message>
+  <message>
+    <source>Return</source>
+    <translation>Retur</translation>
+  </message>
+  <message>
+    <source>&lt;b&gt;Serial number&lt;/b&gt;: %1</source>
+    <translation>&lt;b&gt;Serienummer&lt;/b&gt;: %1</translation>
+  </message>
+  <message>
+    <source>&lt;b&gt;Hardware revision&lt;/b&gt;: %1</source>
+    <translation>&lt;b&gt;maskinvarurevidering&lt;/b&gt;: %1</translation>
+  </message>
+  <message>
+    <source>&lt;b&gt;Firmware version&lt;/b&gt;: %1</source>
+    <translation>&lt;b&gt;Firmware version&lt;/b&gt;: %1</translation>
+  </message>
+  <message>
+    <source> (&lt;font color='#ff6f00'&gt;&lt;i&gt;recommended: %1&lt;/i&gt;&lt;/font&gt;)</source>
+    <translation> (&lt;font color='#ff6f00'&gt;&lt;i&gt;rekommenderas: %1&lt;/i&gt;&lt;/font&gt;)</translation>
+  </message>
+  <message>
+    <source>This component has the recommended firmware version installed.</source>
+    <translation>Denna komponent har den rekommenderade versionen av den inbyggda programvaran installerad.</translation>
+  </message>
+  <message>
+    <source>This component's firmware version does not match the running system firmware version. Printing may not be reliable. Install new firmware now?</source>
+    <translation>Denna komponent's version av den inbyggda programvaran matchar inte versionen av den inbyggda programvaran. Utskrifter kanske inte är tillförlitliga. Installera ny firmware nu?</translation>
+  </message>
+</context>
+<context>
+  <name>InstallingPage</name>
+  <message>
+    <source>Installation complete!</source>
+    <translation>Installationen slutförd!</translation>
+  </message>
+  <message>
+    <source>Yes</source>
+    <translation>Ja</translation>
+  </message>
+  <message>
+    <source>No</source>
+    <translation>Nej</translation>
+  </message>
+  <message>
+    <source>Unpacking installer</source>
+    <translation>Packa upp installationsprogrammet</translation>
+  </message>
+  <message>
+    <source>Extracting %1 to internal storage.</source>
+    <translation>Extraherar %1 till intern lagring.</translation>
+  </message>
+</context>
+<context>
+  <name>BootOptionsPage</name>
+  <message>
+    <source>Cancel</source>
+    <translation>Avbryt</translation>
+  </message>
+  <message>
+    <source>Wipe writable partition</source>
+    <translation>Rensa skrivbar partition</translation>
+  </message>
+  <message>
+    <source>It is possible to unrecoverably and permanently damage your printer with a root console. Never type something into a root console unless you understand what you're typing.</source>
+    <translation>Det är möjligt att oåterkalleligt och permanent skada skrivaren med en rotkonsol. Skriv aldrig in något i en rotkonsol om du inte förstår vad du're skriva.</translation>
+  </message>
+  <message>
+    <source>I'll be careful, I promise</source>
+    <translation>I'kommer att vara försiktig, Jag lovar</translation>
+  </message>
+  <message>
+    <source>&lt;font color="#EEAAAA"&gt;Use this mode with caution.&lt;/font&gt;</source>
+    <translation>&lt;font color="#EEAAAA"&gt;Använd detta läge med försiktighet.&lt;/font&gt;</translation>
+  </message>
+  <message>
+    <source>Fingers are crossed</source>
+    <translation>Fingrar korsas</translation>
+  </message>
+</context>
+<context>
+  <name>VibrationCompPlotter</name>
+  <message>
+    <source>Are you sure you want to remove the data entry from %1?</source>
+    <translation>Är du säker på att du vill ta bort datainmatningen från %1?</translation>
+  </message>
+  <message>
+    <source>X axis</source>
+    <translation>X axel</translation>
+  </message>
+  <message>
+    <source>Y axis</source>
+    <translation>Y axel</translation>
+  </message>
+  <message>
+    <source>Frequency Response Plot: </source>
+    <translation>Frekvensens svarsplats: </translation>
+  </message>
+</context>
+<context>
+  <name>LanAccessPage</name>
+  <message>
+    <source>Return</source>
+    <translation>Tillbaka</translation>
+  </message>
+</context>
+<context>
+  <name>ConsolePage</name>
+  <message>
+    <source>This interface allows you to send G-code commands to the printer. You can enter commands with the virtual keyboard, or put together commands from the shortcut bar at the bottom of the screen. The printer's G-code parser is somewhat picky; here are some tips for how to placate it: </source>
+    <translation>Detta gränssnitt låter dig skicka G-kod kommandon till skrivaren. Du kan ange kommandon med det virtuella tangentbordet, eller sätta ihop kommandon från genvägen längst ner på skärmen. Skrivaren's G-kod parser är något kräsen; här är några tips för hur man blidkar det: </translation>
+  </message>
+  <message>
+    <source>WARNING: It is possible to do permanent, irreversible damage to your printer from a root console. Do not enter commands unless you understand what you are typing.</source>
+    <translation>VARNING: Det är möjligt att göra permanenta, oåterkalleliga skador på skrivaren från en rotkonsol. Ange inte kommandon om du inte förstår vad du skriver.</translation>
+  </message>
+  <message>
+    <source>enter a G-code command</source>
+    <translation>ange ett G-kod-kommando</translation>
+  </message>
+  <message>
+    <source>Home</source>
+    <translation>Hem</translation>
+  </message>
+  <message>
+    <source>Ignore</source>
+    <translation>Ignorera</translation>
+  </message>
+  <message>
+    <source>Close</source>
+    <translation>Stäng</translation>
+  </message>
+  <message>
+    <source>Printer is busy! Are you sure you want to publish this command? Press ignore to hide this message.</source>
+    <translation>Skrivaren är upptagen! Är du säker på att du vill publicera detta kommando? Tryck på ignorera för att dölja detta meddelande.</translation>
+  </message>
+  <message>
+    <source>Export console log</source>
+    <translation>Exportera konsollogg</translation>
+  </message>
+</context>
 </TS>
\ No newline at end of file
