--- printer_ui-orig/printerui/qml/wizard/ConnectPage.qml
+++ printer_ui/printerui/qml/wizard/ConnectPage.qml
@@ -131,9 +131,9 @@
                                     })
                     } else if (modelData.securityEnabled) {
                         dialogStack.push("../settings/WifiSetupPage.qml", { ssid: modelData.ssid, ssidBak: modelData.ssidBak })
-                    } else if(modelData.authenticationType == "EAP"){
+                    } else if(modelData.authenticationType == "EAP" && hasWried){
                         pageStack.push("CertificateConfirm2Page.qml", { ssid: modelData.ssid, ssidBak: modelData.ssidBak })
-                    } else {
+                    } else if (modelData.authenticationType == "OPEN") {
                         NetworkManager.setupWifi(modelData.ssid, modelData.ssid, "OPEN");
                     }
                 }
