--- printer_ui-orig/printerui/qml/main/Home2Page.qml
+++ printer_ui/printerui/qml/main/Home2Page.qml
@@ -1,11 +1,14 @@
+
 import QtQuick 2.12
 import QtQuick.Controls 2.12
 import UIBase 1.0
 import Printer 1.0
+import X1PlusNative 1.0
 
 import ".."
 import "../printer"
 import "../models"
+import "../X1Plus.js" as X1Plus
 import "qrc:/uibase/qml/widgets"
 
 Item {
@@ -17,6 +20,9 @@
     property var carouselImages: WebContents.carouselImages2
     property var defaultImage: DeviceManager.getSystemSetting("device/images_path")
     property bool printPrepare: task.state < PrintTask.RUNNING && task.state > PrintTask.IDLE
+    property var serialNo: DeviceManager.build.seriaNO
+    property var print_png: DeviceManager.getSetting("cfw_print_image", `/mnt/sdcard/x1plus/printers/${serialNo}/images/printing.png`)
+    property var img_print: X1Plus.fileExists(print_png) ? "file://" + print_png: "qrc:/printerui/image/exhibition1.png"
     property var stateIcons: {
         var icons = []
         if (RecordManager.isTimelapseOn) {
@@ -297,7 +303,8 @@
         topMargin: statusPanel.topMargin
         bottomMargin: statusPanel.bottomMargin
         color: hmsItem.visible ? Colors.gray_700: Colors.gray_600
-
+        
+        
         ZCarouselView {
             anchors.fill: parent
             anchors.margins: 10
@@ -305,7 +312,7 @@
             orientation: Qt.Vertical
             visible: !hmsItem.visible && !printStep.visible
             model: carouselImages.length === 0
-                   ? [ "file:" + defaultImage + "exhibition1.png" ]
+                   ? [ img_print ]
                    : carouselImages.map(function(i) { return i.imageUrl })
             onModelChanged: {
                 console.log("Home2Page ZCarouselView image model: ", model)
