--- printer_ui-orig/printerui/qml/main/HomePage.qml
+++ printer_ui/printerui/qml/main/HomePage.qml
@@ -2,6 +2,7 @@
 import QtQuick.Controls 2.12
 import UIBase 1.0
 import Printer 1.0
+import X1PlusNative 1.0
 
 import ".."
 import "../printer"
@@ -11,6 +12,20 @@
 
     property var carouselImages: hmsPanel.visible
                                  ? WebContents.carouselImages2 : WebContents.carouselImages1
+    property var serialNo: DeviceManager.build.seriaNO
+    property var f2: "file://" + DeviceManager.getSetting("cfw_home_image", `/mnt/sdcard/x1plus/printers/${serialNo}/images/home.png`)
+    property var f1: "file://"+ DeviceManager.getSetting("cfw_print_image", `/mnt/sdcard/x1plus/printers/${serialNo}/images/printing.png`)
+    property var img1: imgExists(f1) ? f1: "qrc:/printerui/image/exhibition1.png"
+    property var img2: imgExists(f2) ? f2: "qrc:/printerui/image/exhibition.png"
+ 
+    function imgExists(img){
+        var strPath = img.replace("file://","");
+        if (X1PlusNative.popen(`test -f ${strPath} && echo 1 || echo 0`) == 0){
+            return false;
+        } else {
+            return true;
+        }
+    }
 
     MarginPanel {
         id: showPanel
@@ -28,8 +43,8 @@
             interval: 10000
             orientation: hmsPanel.visible ? Qt.Vertical : Qt.Horizontal
             model: carouselImages.length === 0
-                   ? [ hmsPanel.visible ? "qrc:/printerui/image/exhibition1.png"
-                                        : "qrc:/printerui/image/exhibition.png" ]
+                   ? [ hmsPanel.visible ? img1
+                                        : img2 ]
                    : carouselImages.map(function(i) { return i.imageUrl })
         }
     }
