--- printer_ui-orig/printerui/qml/main/HomePage.qml
+++ printer_ui/printerui/qml/main/HomePage.qml
@@ -2,17 +2,25 @@
 import QtQuick.Controls 2.12
 import UIBase 1.0
 import Printer 1.0
+import X1PlusNative 1.0
 
 import ".."
 import "../printer"
+import "../X1Plus.js" as X1Plus
 import "qrc:/uibase/qml/widgets"
 
 Item {
 
     property var carouselImages: hmsPanel.visible
                                  ? WebContents.carouselImages2 : WebContents.carouselImages1
-    property var defaultImage: DeviceManager.getSystemSetting("device/images_path")
+    property var serialNo: DeviceManager.build.seriaNO
+    property var home_png: DeviceManager.getSetting("cfw_home_image", `/mnt/sdcard/x1plus/printers/${serialNo}/images/home.png`)
+    property var print_png: DeviceManager.getSetting("cfw_print_image", `/mnt/sdcard/x1plus/printers/${serialNo}/images/printing.png`)
+    property var img_print: X1Plus.fileExists(print_png) ? "file://"+print_png: "qrc:/printerui/image/exhibition1.png"
+    property var img_home: X1Plus.fileExists(home_png) ? "file://"+home_png: "qrc:/printerui/image/exhibition.png" 
     EventTrack.pageName: "Home"
+    
+    
 
     MarginPanel {
         id: showPanel
@@ -30,8 +38,8 @@
             interval: 10000
             orientation: hmsPanel.visible ? Qt.Vertical : Qt.Horizontal
             model: carouselImages.length === 0
-                   ? [ hmsPanel.visible ? "file:" + defaultImage + "exhibition1.png"
-                                        : "file:" + defaultImage + "exhibition.png" ]
+                   ? [ hmsPanel.visible ? img_print
+                                        : img_home ]
                    : carouselImages.map(function(i) { return i.imageUrl })
             onModelChanged: {
                 console.log("HomePage ZCarouselView image model: ", model)
