--- printer_ui-orig/printerui/qml/printer/AdjustPage.qml
+++ printer_ui/printerui/qml/printer/AdjustPage.qml
@@ -194,9 +194,6 @@
                     icon: "../../icon/" + modelData.tip + ".svg"
                     text: modelData.name
                     onClicked: {
-                        if ((PrintManager.homedState & PrintManager.AXIS_HOMED_ALL) !== PrintManager.AXIS_HOMED_ALL) {
-                            limitFrame()
-                        }
                         axisController.moveAxis(modelData.axis, modelData.dir, modelData.unit)
                     }
 
@@ -749,9 +746,6 @@
                 textColor: StateColors.get("white_900_checked_disabled")
                 radius: 239
                 onClicked: {
-                    if ((PrintManager.homedState & PrintManager.AXIS_HOMED_ALL) != PrintManager.AXIS_HOMED_ALL) {
-                        limitFrame()
-                    }
                     axisController.moveAxis(modelData.axis, modelData.dir, AxisController.MM_10)
                 }
             }
@@ -763,9 +757,6 @@
                 textColor: StateColors.get("gray_400_checked_disabled")
                 radius: 133
                 onClicked: {
-                    if ((PrintManager.homedState & PrintManager.AXIS_HOMED_ALL) != PrintManager.AXIS_HOMED_ALL) {
-                        limitFrame()
-                    }
                     axisController.moveAxis(modelData.axis, modelData.dir, AxisController.MM)
                 }
             }
