--- printer_ui-orig/printerui/qml/printer/AdjustPage.qml
+++ printer_ui/printerui/qml/printer/AdjustPage.qml
@@ -16,7 +16,9 @@
     property var extruders: hasExtruderList ? extruderList.extruders : []
     property var currentExtruder: hasExtruderList ? extruderList.currentExtruder : null
     property var tmpModelArray: hasExtruderList ? (extruders.map(n=>n.extruderHeater)).concat(tempModel) : [heaters.hotend].concat(tempModel)
-
+    property var isHomed: (PrintManager.homedState & PrintManager.AXIS_HOMED_ALL) >= PrintManager.AXIS_HOMED_ALL
+    property bool ignoreDialog: false
+    
     property var speedAdjust: QtObject {
         property string name: "speed"
         property int currentTemp: PrintManager.currentTask.printSpeed
@@ -153,18 +155,32 @@
                 model: xyAsixModel
                 delegate: axisComp
             }
-
-            ZButton {
+            Rectangle {
                 id: axisHome
                 width: 103
                 height: 103
                 anchors.centerIn: parent
-                type: ZButtonAppearance.Tertiary
-                radius: 44
-                iconSize: 0
-                icon: "../../icon/vector.svg"
-                onClicked: {
-                    axisController.moveCenter(PrintManager.currentTask.stage === PrintTask.PAUSED)
+                radius: width/2
+                color:  "#626260"
+                    ZImage {
+                        id: axisHomeImage
+                        width: 70 
+                        height: 70
+                        anchors.centerIn: parent
+                        fillMode: Image.PreserveAspectFit 
+                        originSource: (isHomed) ? "../../icon/components/home.svg" : "../../icon/vector.svg"
+                        tintColor: (isHomed) ?  Colors.brand : "#eb6600"
+                        Behavior on tintColor {
+                            ColorAnimation { duration: 500 } 
+                        }
+                    }
+                MouseArea {
+                    anchors.fill:parent
+                    onClicked: {
+                        axisController.moveCenter(PrintManager.currentTask.stage === PrintTask.PAUSED)
+                    }
+                    onEntered: axisHome.opacity = 0.8
+                    onExited: axisHome.opacity = 1.0
                 }
             }
         }
@@ -210,7 +226,7 @@
                     }
 
                     onClicked: {
-                        if ((PrintManager.homedState & PrintManager.AXIS_HOMED_ALL) !== PrintManager.AXIS_HOMED_ALL) {
+                        if (!isHomed && !ignoreDialog) {
                             limitFrame()
                         }
                         axisController.moveAxis(modelData.axis, modelData.dir, modelData.unit)
@@ -828,7 +844,7 @@
                 EventTrack.itemName: "bigSector_" + modelData.name
 
                 onClicked: {
-                    if ((PrintManager.homedState & PrintManager.AXIS_HOMED_ALL) != PrintManager.AXIS_HOMED_ALL) {
+                    if (!isHomed && !ignoreDialog) {
                         limitFrame()
                     }
                     axisController.moveAxis(modelData.axis, modelData.dir, AxisController.MM_10)
@@ -846,7 +862,7 @@
                 EventTrack.itemName: "smallSector_" + modelData.name
 
                 onClicked: {
-                    if ((PrintManager.homedState & PrintManager.AXIS_HOMED_ALL) != PrintManager.AXIS_HOMED_ALL) {
+                    if (!isHomed && !ignoreDialog) {
                         limitFrame()
                     }
                     axisController.moveAxis(modelData.axis, modelData.dir, AxisController.MM)
@@ -901,11 +917,13 @@
         dialogStack.popupDialog(
                     "TextConfirm", {
                         name: "Limit Frame",
-                        titles: [qsTr("Home"), qsTr("Close")],
+                        type: TextConfirm.YES_NO_CANCEL,
+                        titles: [qsTr("Home"), qsTr("Ignore"), qsTr("Close")],
                         text: qsTr("Before any movement, please home all axes to find the boundary of build volume."),
                         onYes: function() {
                             axisController.moveCenter(PrintManager.currentTask.stage === PrintTask.PAUSED)
-                        }
+                        },
+                        onNo: function() {ignoreDialog = true},
                     })
 
     }
