--- printer_ui-orig/printerui/qml/printer/CalibrationPage.qml
+++ printer_ui/printerui/qml/printer/CalibrationPage.qml
@@ -19,20 +19,31 @@
         id: usagePanel
         width: 677
         anchors.top: parent.top
-        topMargin: 60
+        topMargin: 26
         anchors.bottom: parent.bottom
-        bottomMargin: 57
+        bottomMargin: 26
         anchors.left: parent.left
-        leftMargin: 15
+        leftMargin: 26
 
         Text {
-            x: 25
-            y: 40
+            id: guideLabel
+            anchors.horizontalCenter: parent.horizontalCenter
+            anchors.top: parent.top
+            anchors.topMargin: 20
             font: Fonts.head_36
             color: Colors.brand
             text: qsTr("Calibration Step Selection")
         }
 
+        ZLineSplitter {
+            id: stepSplit2
+            anchors.top: guideLabel.bottom
+            anchors.topMargin: 10
+            alignment: Qt.AlignTop
+            padding: 15
+            color: Colors.brand
+        }
+
         Repeater {
             model: calibraSelect
 
@@ -66,17 +77,17 @@
 
     MarginPanel {
         id: stepPanel
-        height: 494
+        height: 585
         anchors.top: parent.top
         anchors.left: usagePanel.right
         anchors.right: parent.right
-        leftMargin: 20
-        topMargin: 60
-        rightMargin: 15
+        leftMargin: 14
+        topMargin: 26
+        rightMargin: 26
 
         Text {
             x: 23
-            y: 29
+            y: 24
             font: Fonts.head_28
             color: Colors.brand
             text: qsTr("Calibration Flow")
@@ -85,7 +96,7 @@
         ZLineSplitter {
             id: stepSplit
             alignment: Qt.AlignTop
-            offset: 80
+            offset: 70
             padding: 15
             color: Colors.brand
         }
@@ -110,10 +121,10 @@
         anchors.right: parent.right
         anchors.top: stepPanel.bottom
         anchors.bottom: parent.bottom
-        bottomMargin: 57
-        leftMargin: 20
-        topMargin: 10
-        rightMargin: 13
+        bottomMargin: 26
+        leftMargin: 14
+        topMargin: 14
+        rightMargin: 26
 
         ZButton {
             id: startButton
@@ -141,17 +152,9 @@
         }
     }
 
-    ZButton {
-        x: 18
-        y: 15
-        height: width
-        width: 88
-        cornerRadius: width / 2
-        iconSize: -1
-        type: ZButtonAppearance.Secondary
-        icon: "../../icon/return.svg"
+    X1PBackButton {
         onClicked: {
-            calibration.parent.pop()
+            pageStack.pop()
         }
     }
 }
