--- printer_ui-orig/printerui/qml/printer/Choise.qml
+++ printer_ui/printerui/qml/printer/Choise.qml
@@ -15,6 +15,9 @@
     property var selectCount: 7.5
 
     id: comboBox
+
+    signal choiseTapped(); // "activated" signal cannot use, use this insted
+
     height: 60
     background: Rectangle { visible: !readOnly; color: backgroundColor; radius: 15; border.width: 0 }
     contentItem: ZText {
@@ -55,7 +58,7 @@
             font: listTextFont
             color: Colors.gray_100
             padding: 16
-//            maxWidth: comboBox.width
+            maxWidth: width < contentWidth ? parent.width : 0
             verticalAlignment: Text.AlignVCenter
             text: modelData
             TapHandler {
@@ -63,6 +66,7 @@
                 onTapped: {
                     comboBox.currentIndex = index
                     comboBox.popup.close()
+                    choiseTapped()
                 }
             }
             HoverHandler {
