--- printer_ui-orig/printerui/qml/printer/DryFilamentPage.qml
+++ printer_ui/printerui/qml/printer/DryFilamentPage.qml
@@ -46,7 +46,7 @@
         anchors.left: parent.left
         anchors.top:  parent.top
         leftMargin: 26
-        topMargin: 44
+        topMargin: 26
 
         Image {
             anchors.centerIn: parent
@@ -448,15 +448,7 @@
         }
     }
 
-    ZButton {
-        x: 18
-        y: 24
-        height: width
-        width: 88
-        cornerRadius: width / 2
-        iconSize: -1
-        type: ZButtonAppearance.Secondary
-        icon: "../../icon/return.svg"
+    X1PBackButton {
         onClicked: { pageStack.pop() }
     }
 }
