--- printer_ui-orig/printerui/qml/printer/FilamentPage.qml
+++ printer_ui/printerui/qml/printer/FilamentPage.qml
@@ -13,6 +13,7 @@
                                 ? amsModel[amsList.currentIndex] : null
     property var trayModel: currentFeeder ? currentFeeder.trays : []
     property bool hasAms: feeder.hasAms
+    //property bool hasAms: true
     property bool loading: feeder.loading
     property var currentTray: feeder.currentTray
     property int loadStep: feeder.loadStep
@@ -388,7 +389,7 @@
             }
         }
     }
-
+    
     MarginPanel {
         id: howToPanel
         height: 538
@@ -651,36 +652,49 @@
                                               : qsTr("Empty")
                     }
 
-                    ZImage {
+                    Rectangle {
                         width: parent.width
-                        height: 60
-                        y: 96
+                        height: parent.height/2.75
                         anchors.horizontalCenter: parent.horizontalCenter
-                        fillMode: Image.Pad
-                        tintColor: {
-                            var color = colorHeight.y >= (y + height / 2) ? "transparent" : (colorHeight.alpha !== 0 ? colorHeight.color : Colors.white_900)
-                            var color2 = Colors.blend([color, parent.color, Colors.gray_800])
-
-                            if (modelData.color.a > 0 && modelData.color.a < 1)
-                                return Colors.white_900
-                            else
-                                return Colors.different(color2)
-                        }
-                        originSource: modelData.thirdparty
-                                ? "../../icon/edit2.svg" : "../../icon/show_eye.svg"
+                        anchors.bottom: parent.bottom
+                        radius: parent.radius
+                        opacity: 0.8
+                        layer.enabled: true
+                        color: Colors.gray_800
                         visible: modelData.exist
+                        Rectangle {
+                            height: parent.radius
+                            anchors.top: parent.top
+                            anchors.left: parent.left
+                            anchors.right: parent.right
+                            color: parent.color
+                        }
+                        
+                        ZImage {
+                            width: 60
+                            height: 60
+                            anchors.verticalCenter: parent.verticalCenter
+                            anchors.horizontalCenter: parent.horizontalCenter
+                            fillMode: Image.Pad
+                            tintColor: {
+                                var color = colorHeight.y >= (y + height / 2) ? "transparent" : (colorHeight.alpha !== 0 ? colorHeight.color : Colors.white_900)
+                                var color2 = Colors.blend([color, parent.color, Colors.gray_800])
+
+                                if (modelData.color.a > 0 && modelData.color.a < 1)
+                                    return Colors.white_900
+                                else
+                                    return Colors.different(color2)
+                            }
+                            originSource: modelData.thirdparty
+                                    ? "../../icon/edit2.svg" : "../../icon/show_eye.svg"
+                        }
+
                         TapHandler {
                             onTapped: {
                                 if (loading) return;
-                                if (trayItem.ListView.isCurrentItem) {
-                                    if (!modelData.exist)
+                                if (!modelData.exist)
                                         return
-                                    requireEdit(modelData)
-                                } else {
-                                    trayItem.ListView.view.currentIndex = index
-                                    feeder.selectTray(modelData)
-                                    selectTray = modelData
-                                }
+                                requireEdit(modelData)
                             }
                         }
                     }
