--- printer_ui-orig/printerui/qml/printer/PrintListener.qml
+++ printer_ui/printerui/qml/printer/PrintListener.qml
@@ -39,6 +39,10 @@
                 }
             }
 
+            if (name == "update" && screenSaver.overrideUpdater) {
+                // suppress "The version has been updated to..." message
+                return
+            }
             dialogStack.popupDialog("ErrorPrompt", {
                                         level: type,
                                         name: name,
