--- printer_ui-orig/printerui/qml/settings/CarbonFunction.qml
+++ printer_ui/printerui/qml/settings/CarbonFunction.qml
@@ -44,7 +44,7 @@
                                     titles: [qsTr("Later"), qsTr("Try it Now")],
                                     callback: function(index) {
                                         if (index === TextConfirm.NO) {
-                                            Printer.jumpTo("Printer/Utilities")
+                                            Printer.jumpTo("Settings/Maintenance")
                                             stack3.push("../printer/CalibrationPage.qml")
                                         }
                                     }
