--- printer_ui-orig/printerui/qml/settings/CertificateConfirm1Page.qml
+++ printer_ui/printerui/qml/settings/CertificateConfirm1Page.qml
@@ -244,7 +244,7 @@
                     // no need input password when certificate is CA type
                     if (modelData.installState) {
                         dialogStack.popupDialog("TextConfirm", {
-                                                    name: "uninstall certificate id:" + modelData.certificateId,
+                                                    name: "uninstall certificate id",
                                                     text: qsTr("Do you want to delete %1?").arg(modelData.certificateId),
                                                     defaultButton: TextConfirm.NO,
                                                     onYes: function(index) {
