--- printer_ui-orig/printerui/qml/settings/NetworkPage.qml
+++ printer_ui/printerui/qml/settings/NetworkPage.qml
@@ -6,7 +6,8 @@
 import "qrc:/uibase/qml/widgets"
 import ".."
 
-Item {
+Rectangle {
+    color: Colors.gray_700
 
     property var network: NetworkManager.network
     property bool wifiIsOn: network.isOn && network.powerState
@@ -29,10 +30,16 @@
 
     MarginPanel {
         id: wifiPanel
-        width: (parent.width - 24) / 2
-        height: parent.height
-        anchors.bottomMargin: 24
+        anchors.top: parent.top
+        anchors.left: parent.left
+        anchors.right: parent.horizontalCenter
+        leftMargin: 10
+        rightMargin: 7
+        anchors.bottom: parent.bottom
+        bottomMargin: 10
+        radius: 15
         visible: hasNetworkSubCard
+        color: Colors.gray_800
 
         Item {
             id: wlanTitle
@@ -281,13 +288,16 @@
 
     MarginPanel {
         id: rightPanel
-        width: (parent.width - 24) / 2
-        height: parent.height
+        anchors.top: parent.top
         anchors.left: wifiPanel.right
-        leftMargin: 24
+        leftMargin: 14
         anchors.right: parent.right
-        rightMargin: 16
-        bottomMargin: 24
+        rightMargin: 10
+        anchors.bottom: parent.bottom
+        bottomMargin: 10
+        radius: 15
+        color: Colors.gray_800
+
         visible: hasNetworkSubCard
 
         Item {
