--- printer_ui-orig/printerui/qml/settings/NetworkPage.qml
+++ printer_ui/printerui/qml/settings/NetworkPage.qml
@@ -5,18 +5,21 @@
 
 import "qrc:/uibase/qml/widgets"
 import ".."
+import "../X1Plus.js" as X1Plus
 
-Item {
+Rectangle {
+    color: Colors.gray_700
 
     property var network: NetworkManager.network
     property bool wifiIsOn: network.isOn && network.powerState
-    property var wiredNetwork: NetworkManager.wiredNetwork
+    property var wiredNetwork: X1Plus.Network.wiredNetwork()
     property bool wiredIsOn: wiredNetwork.isOn && wiredNetwork.powerState
     property var wifiNetwork: NetworkManager.wifiNetwork
     property var wifiListModel: NetworkManager.wifiList
     property bool hasNetworkSubCard: NetworkManager.networkSubCardState
-    property bool hasWried: NetworkManager.wiredNetwork !== undefined
+    property bool hasWried: X1Plus.Network.wiredNetwork() !== undefined
     property bool hasCertificate: NetworkManager.certificateList !== null
+    property bool hasExpansion: X1Plus.Expansion.status() !== null
 //    property var currentNetwork: NetworkManager.currentRoute
 
     property bool visible2: visible // avoid initial change to false
@@ -29,10 +32,16 @@
 
     MarginPanel {
         id: wifiPanel
-        width: (parent.width - 24) / 2
-        height: parent.height
-        anchors.bottomMargin: 24
+        anchors.top: parent.top
+        anchors.left: parent.left
+        anchors.right: parent.horizontalCenter
+        leftMargin: 10
+        rightMargin: 7
+        anchors.bottom: parent.bottom
+        bottomMargin: 10
+        radius: 15
         visible: hasNetworkSubCard
+        color: Colors.gray_800
 
         Item {
             id: wlanTitle
@@ -281,13 +290,16 @@
 
     MarginPanel {
         id: rightPanel
-        width: (parent.width - 24) / 2
-        height: parent.height
+        anchors.top: parent.top
         anchors.left: wifiPanel.right
-        leftMargin: 24
+        leftMargin: 14
         anchors.right: parent.right
-        rightMargin: 16
-        bottomMargin: 24
+        rightMargin: 10
+        anchors.bottom: parent.bottom
+        bottomMargin: 10
+        radius: 15
+        color: Colors.gray_800
+
         visible: hasNetworkSubCard
 
         Item {
@@ -368,10 +380,37 @@
                     color: wiredNetwork.powerState ? Colors.brand : Colors.warning
                 }
             }
+            
+            Item {
+                id: x1PlusLogoItem
+                visible: hasExpansion
+                anchors.left: wiredNetworkItem.left
+                anchors.top: wiredNetworkItem.bottom
+                anchors.topMargin: -19
+                width: 468
+                height: 48
+                
+                ZImage {
+                    id: x1PlusLogo
+                    anchors.left: x1PlusText.right
+                    anchors.leftMargin: 5
+                    width: 48
+                    height: 48
+                    originSource: "../../icon/components/cfw.png"
+                }
+                
+                Text {
+                    id: x1PlusText
+                    font: Fonts.body_28
+                    anchors.verticalCenter: x1PlusLogo.verticalCenter
+                    text: X1Plus.Expansion.productName()
+                    color: Colors.gray_300
+                }
+            }
 
             Item {
                 id: connectItem
-                anchors.top: wiredNetworkItem.bottom
+                anchors.top: hasExpansion ? x1PlusLogoItem.bottom : wiredNetworkItem.bottom
                 anchors.bottom: line3.top
                 anchors.left: wiredNetworkItem.left
                 width: 468
@@ -401,7 +440,7 @@
             ZLineSplitter {
                 id: line3
                 anchors.top: parent.top
-                anchors.topMargin: 160
+                anchors.topMargin: 193
                 alignment: Qt.AlignTop
                 padding: 36
                 visible: wiredIsOn
@@ -439,7 +478,7 @@
 
             Item {
                 id: infoPanel2
-                anchors.top: line4.bottom
+                anchors.top: hasCertificate ? line4.bottom : line3.bottom
                 height: 132
                 anchors.left: wiredNetworkItem.left
                 anchors.right: wiredNetworkItem.right
