--- printer_ui-orig/printerui/qml/assist/SelfTestPage.qml
+++ printer_ui/printerui/qml/assist/SelfTestPage.qml
@@ -18,7 +18,46 @@
         id: usagePanel
         width: 677
         height: parent.height
+        anchors.top: parent.top
+        anchors.bottom: parent.bottom
+        anchors.left: parent.left
+        leftMargin: 26
+        topMargin: 26
+        bottomMargin: 26
+        
+        Text {
+            id: guideLabel
+            anchors.horizontalCenter: parent.horizontalCenter
+            anchors.top: parent.top
+            anchors.topMargin: 20
+            font: Fonts.head_36
+            color: Colors.brand
+            text: usageTexts[0][0]
+        }
+
+        ZLineSplitter {
+            id: stepSplit2
+            anchors.top: guideLabel.bottom
+            anchors.topMargin: 10
+            alignment: Qt.AlignTop
+            padding: 15
+            color: Colors.brand
+        }
 
+        Text {
+            anchors.left: parent.left
+            anchors.leftMargin: 20
+            anchors.right: parent.right
+            anchors.rightMargin: 20
+            anchors.top: stepSplit2.bottom
+            anchors.topMargin: 30
+            font: Fonts.body_30
+            color: Colors.gray_300
+            wrapMode: Text.WordWrap
+            text: usageTexts[0][1]
+        }
+        
+        /*
         Repeater {
             model: usageTexts
 
@@ -42,14 +81,19 @@
                 }
             }
         }
+        */
     }
 
     MarginPanel {
         id: stepPanel
-        height: 494
+        height: 585
+        anchors.top: parent.top
         anchors.left: usagePanel.right
         anchors.right: parent.right
-        leftMargin: 20
+        leftMargin: 14
+        topMargin: 26
+        bottomMargin: 26
+        rightMargin: 26
 
         Text {
             x: 23
@@ -88,8 +132,10 @@
         anchors.right: parent.right
         anchors.top: stepPanel.bottom
         anchors.bottom: parent.bottom
-        leftMargin: 20
-        topMargin: 10
+        bottomMargin: 26
+        leftMargin: 14
+        topMargin: 14
+        rightMargin: 26
 
         ZButton {
             id: startButton
@@ -103,4 +149,10 @@
             }
         }
     }
+
+    X1PBackButton {
+        onClicked: {
+            pageStack.pop()
+        }
+    }
 }
